### PR TITLE
Add new overload to PersistentDataContainer#has

### PR DESCRIPTION
**Pull request by u9g**
Adds the new overload: `PersistentDataContainer#has(NamespacedKey key)`

### CODE REVIEW

File: patches/api/0352-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      */
+     @NotNull
+     PersistentDataAdapterContext getAdapterContext();
++
++    // Paper start
++    /**
++     * Returns if the persistent metadata provider has metadata registered
++     * matching the provided key.
++     * 
++     * @param key the key to look up in the custom tag map
```
Comment (lynxplay) [inline patches/api/0352-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch]: The concept of a custom tag map is pretty unknown to the API.
This should probably be rephrased to something along the lines of
```
@param key the key  for which existence should be checked.
```

File: patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+diff --git a/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java b/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
+index 2c59f09a9261a1690951161fd856a5848d9885b7..987c8144738c6e016f5b60b41806bb2a9afbd1a2 100644
+--- a/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
++++ b/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
+@@ -161,5 +161,15 @@ public class CraftPersistentDataContainer implements PersistentDataContainer {
+     public void clear() {
+         this.customDataTags.clear();
+     }
++
++    public boolean has(@NotNull NamespacedKey key) {
```
Comment (lynxplay) [inline patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch]: I believe we do infact want the `@Override` annotation here.

File: patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
++        Validate.notNull(key, "The provided key for the custom value was null");
++
++        Tag value = this.customDataTags.get(key.toString());
```
Comment (lynxplay) [inline patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch]: This entire section can be simplified to
```java
return this.customDataTags.has(key.toString());
```

**Review by lynxplay — CHANGES_REQUESTED** [review]

File: patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch (hunk 2 above)
Comment (u9g) [reply patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch]: Done :^)

File: patches/api/0352-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch (hunk 1 above)
Comment (u9g) [reply patches/api/0352-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch]: Done :D

File: patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch (hunk 3 above)
Comment (u9g) [reply patches/server/0842-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch]: Done ;D

**Review by Machine-Maker — APPROVED** [review]
Looks good. Is there an existing patch this might fit into? Kinda small for its own patch, but if there isn’t, it’s fine.

**Comment by lynxplay** [comment]
Was looking through it, no patch yet that touches the PDC API, so this might be a good first other changes can append into later.

**Review by lynxplay — APPROVED** [review]

**Review by kennytv — APPROVED** [review]

**Review by JRoy — APPROVED** [review]

**Pull request merged**